designs/helm-charts/expose-container-images.md

```diff
@@ -0,0 +1,92 @@
+---
+oep-number: OEP 1
+title: Expose OpenEBS HelmChart's Container Images
+authors:
+  - "@tiagolobocastro"
+owners:
+  - "@tiagolobocastro"
+editor: TBD
+creation-date: 2024-10-03
+last-updated: 2024-10-03
+status: provisional
+---
+
+# Expose OpenEBS HelmChart's Container Images
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [User Stories [optional]](#user-stories-optional)
+      * [Story 1](#story-1)
+      * [Story 2](#story-2)
+    * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Drawbacks [optional]](#drawbacks-optional)
+* [Alternatives [optional]](#alternatives-optional)
+
+## Summary
+
+This proposal aims to add a comprehensive list of all possible container images required by OpenEBS charts. This will help users and developers to easily identify and pull the necessary images for deploying OpenEBS.
+
+## Motivation
+
+The motivation behind this proposal is to simplify the deployment process of OpenEBS by providing a clear and concise list of all required container images. This will reduce the time and effort needed to search for and pull the correct images, thereby improving the user experience.
+Another OEP can be built on top of this, for creating an easily downloadable tar file, comprised of all required container images.
+
+### Goals
+
+Providing a detailed list of all container images required by each OpenEBS chart.
+An automated mechanism must be in place, ensuring the list is kept up-to-date with any changes in the charts.
+
+### Non-Goals
+
+Creating a downloadable bundle of all container images (to be addressed by another OEP)
+Building or maintaining container images.
+Correlating images to specific functionality (ie may contain images a given user would not required for their configuration).
+
+## Proposal
+
+The proposal is to add an annotation images in the Chart.yaml file of each OpenEBS chart. This annotation will contain a list of all container images required by the chart. The list will be automatically updated to reflect any changes in the charts, when those changes are made.
+Each OpenEBS chart will use the annotation from the charts it depends on, thus facilitating and relegating some of the work to the respective chart
```

Comment:
Are the external dependent charts following this annotation?

Reply:
Some yes, example: https://github.com/bitnami/charts/blob/08d9c3e69a2089a076542b894a7aee96e42c0320/bitnami/etcd/Chart.yaml#L7
But we can't be sure they'd all do this, so some amount of grafting may be required for those who don't